mono/eglib/gstr.c

```diff
@@ -46,6 +46,8 @@
 #if (_POSIX_C_SOURCE >= 200112L) && !_GNU_SOURCE
 #define USE_STRERROR_R_XSI
 #endif
+#elif defined(__APPLE__)
```

Comment:
It seems to be not just Apple but probably the whole of Unix ([`strerror_r` at FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=strerror&apropos=0&sektion=3&manpath=FreeBSD+11.1-RELEASE+and+Ports&arch=default&format=html)) ... the macro should probably be defined the other way around: only use the GNU-specific version if on Linux.